tests/test_basic_ops_dict.py

```diff
@@ -398,17 +443,162 @@ def test_dict_pop_error(cfg: Dict[Any, Any], key: Any, expectation: Any) -> None
             "incompatible_key_type",
             False,
         ),
+        (
+            DictConfig(content={Enum1.FOO: "foo"}, key_type=Enum1, element_type=str),
+            "FOO",
+            True,
+        ),
+        (
+            DictConfig(content={Enum1.FOO: "foo"}, key_type=Enum1, element_type=str),
+            None,
+            False,
+        ),
+        # int key type
         ({1: "a", 2: {}}, 1, True),
         ({1: "a", 2: {}}, 2, True),
         ({1: "a", 2: {}}, 3, False),
         ({1: "a", 2: "???"}, 2, False),
+        ({1: "a", 2: "???"}, None, False),
+        ({1: "a", 2: "???"}, "1", False),
+        # (DictConfig({1: "a", 2: "???"}, key_type=int), "1", True),
+        # float key type
+        ({1.1: "a", 2.2: {}}, 1.1, True),
+        ({1.1: "a", 2.2: {}}, "1.1", False),
+        (DictConfig({1.1: "a", 2.2: {}}, key_type=float), "1.1", False),
+        ({1.1: "a", 2.2: {}}, 2.2, True),
+        ({1.1: "a", 2.2: {}}, 3.3, False),
+        ({1.1: "a", 2.2: "???"}, 2.2, False),
+        ({1.1: "a", 2.2: "???"}, None, False),
+        # bool key type
+        ({True: "a", False: {}}, True, True),
+        ({True: "a", False: {}}, False, True),
+        ({True: "a", False: {}}, "no", False),
+        ({True: "a", False: {}}, 1, True),
+        ({True: "a", False: {}}, None, False),
+        ({True: "a", False: "???"}, False, False),
     ],
 )
 def test_in_dict(conf: Any, key: str, expected: Any) -> None:
     conf = OmegaConf.create(conf)
     assert (key in conf) == expected
 
 
+@pytest.mark.parametrize(
+    "cfg_key_type,data",
+    [
+        (str, {"a": 10}),
+        (int, {1: "a"}),
+        (float, {123.45: "a"}),
+        (bool, {True: "a"}),
+        (Enum1, {Enum1.FOO: "foo"}),
+    ],
+)
+@pytest.mark.parametrize(
+    "key",
+    ["a", "b", 1, 2, 123.45, 67.89, True, False, Enum1.FOO, Enum1.BAR, "FOO", "BAR"],
+)
+class TestDunderMethodsAgainstStandardPythonDict:
+    @pytest.fixture
+    def python_dict(self, data: Dict[Any, Any]) -> Dict[Any, Any]:
+        return data.copy()
+
+    @pytest.fixture
+    def cfg(self, python_dict: Any) -> DictConfig:
+        return DictConfig(content=python_dict)
+
+    @pytest.fixture
+    def cfg_typed(self, python_dict: Any, cfg_key_type: Any) -> DictConfig:
+        return DictConfig(content=python_dict, key_type=cfg_key_type)
+
+    @pytest.fixture
+    def key_coerced(self, key: Any, cfg_key_type: Any) -> Any:
```

Comment:
Is pytest passing other parameters from the function this fixture is used in automatically?!

Reply:
I will probably be simpler to test the dicts with the special enum key in a different class/function instead of making everything more complicated to support them in this test.

Reply:
> Is pytest passing other parameters from the function this fixture is used in automatically?!

Not sure what you mean? When `key_coerced` is used in the `test__setitem__typed` function, the value of `key` that was used to create the fixture value of `key_coerced` is the same as the value of `key` that is passed in to the `test__setitem__typed` function.

Reply:
> I will probably be simpler to test the dicts with the special enum key in a different class/function instead of making everything more complicated to support them in this test.

Ok, I'll refactor.

Reply:
> > Is pytest passing other parameters from the function this fixture is used in automatically?!
> 
> Not sure what you mean? When `key_coerced` is used in the `test__setitem__typed` function, the value of `key` that was used to create the fixture value of `key_coerced` is the same as the value of `key` that is passed in to the `test__setitem__typed` function.

So key is coming from the class parametrize, but where is cfg_key_type coming from?

Reply:
> So key is coming from the class parametrize, but where is cfg_key_type coming from?

The class `TestDunderMethodsAgainstStandardPythonDict` has two `@parametrize` decorators:
```
@pytest.mark.parametrize(
    "cfg_key_type,data",
    [
        (str, {"a": 10}),
        (int, {1: "a"}),
        (float, {123.45: "a"}),
        (bool, {True: "a"}),
        (Enum1, {Enum1.FOO: "foo"}),
    ],
)
@pytest.mark.parametrize(
    "key",
    ["a", "b", 1, 2, 123.45, 67.89, True, False, Enum1.FOO, Enum1.BAR, "FOO", "BAR"],
)
class TestDunderMethodsAgainstStandardPythonDict:
    ....
```

This means that every possible combination of `cfg_key_type` and `key` is used:
```
  (cfg_key_type, key) == (str, "a"),
  (cfg_key_type, key) == (str, "b"),
  (cfg_key_type, key) == (str, 1),
  (cfg_key_type, key) == (str, 1),
  ...
  (cfg_key_type, key) == (int, "a"),
  (cfg_key_type, key) == (int, "b"),
  (cfg_key_type, key) == (int, 1),
  (cfg_key_type, key) == (int 1),
  ...
  (cfg_key_type, key) == (Enum1, "a"),
  (cfg_key_type, key) == (Enum1, "b"),
  (cfg_key_type, key) == (Enum1, 1),
  (cfg_key_type, key) == (Enum1 1),
  ...
```